src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8942,6 +8990,14 @@ private MethodSymbol CheckMethodGroupReceiverNullability(BoundMethodGroup group,
 
         public override BoundNode? VisitLambda(BoundLambda node)
         {
+            if (_visitingConversionOperand != node)
+            {
+                VisitLambda(node, delegateTypeOpt: null);
+#if DEBUG
+                Debug.Assert(_errorLambdas.Add(node), $"Lambda visited again without conversion: {node.GetDebuggerDisplay()}");
+#endif
+            }
```

Comment:
Should we return here? The comment below doesn't seem applicable to this code path.

Reply:
The comment might need updating, but the `SetNotNullResult(node);` call is still needed - it indicates that the lambda itself is not null. The `private VisitLambda` overload doesn't do that.